awx/main/models/credential/injectors.py

```diff
@@ -96,25 +97,25 @@ def _openstack_data(cred):
 
 
 def openstack(cred, env, private_data_dir):
-    handle, path = tempfile.mkstemp(dir=private_data_dir)
+    handle, path = tempfile.mkstemp(dir=os.path.join(private_data_dir, 'env'))
     f = os.fdopen(handle, 'w')
     openstack_data = _openstack_data(cred)
     yaml.safe_dump(openstack_data, f, default_flow_style=False, allow_unicode=True)
     f.close()
     os.chmod(path, stat.S_IRUSR | stat.S_IWUSR)
     # TODO: constant for container base path
-    env['OS_CLIENT_CONFIG_FILE'] = os.path.join('/runner', os.path.basename(path))
+    env['OS_CLIENT_CONFIG_FILE'] = os.path.join('/runner', 'env', os.path.basename(path))
 
 
 def kubernetes_bearer_token(cred, env, private_data_dir):
     env['K8S_AUTH_HOST'] = cred.get_input('host', default='')
     env['K8S_AUTH_API_KEY'] = cred.get_input('bearer_token', default='')
     if cred.get_input('verify_ssl') and 'ssl_ca_cert' in cred.inputs:
         env['K8S_AUTH_VERIFY_SSL'] = 'True'
-        handle, path = tempfile.mkstemp(dir=private_data_dir)
+        handle, path = tempfile.mkstemp(dir=os.path.join(private_data_dir, 'env'))
         with os.fdopen(handle, 'w') as f:
             os.chmod(path, stat.S_IRUSR | stat.S_IWUSR)
             f.write(cred.get_input('ssl_ca_cert'))
-        env['K8S_AUTH_SSL_CA_CERT'] = os.path.join('/runner', os.path.basename(path))
+        env['K8S_AUTH_SSL_CA_CERT'] = os.path.join('/runner', 'env', os.path.basename(path))
```

Comment:
I'm much less happy with these than I thought I would be.

There are tons of places in code where we put something in a directory, and other part of code relies on knowledge of how the directory & naming of that file was done. I would like something more robust to changes, and harder to get wrong for the host / container paths.

Reply:
So, to illustrate this point, let me write logic similar to what I've done in tests here

```python
env['K8S_AUTH_SSL_CA_CERT'] = path.replace(private_data_dir, '/runner')
```

Now, maybe this would be best implemented in a utility method (for better future control and error handling).

In general terms, this is where I'm interested in going, and I believe something like this would be hugely valuable for maintainability.